### ISSUE TITLE
Update icepack version

### ISSUE THREAD

**Issue by shlyaeva** [JCSDA-internal/soca#1119]
It would be good to update icepack version used with soca to a more recent one (e.g. one that's currently used in UFS).

I looked into what changes are made in this fork in the tag that [EMC is currently using](https://github.com/JCSDA-internal/Icepack/commits/73136ee8dcdbe378821e540488a5980a03d8abe6/), and the only needed changes are for ecbuild, so updating technically should be fairly easy (pick a commit off the CICE-Consortium/Icepack and cherry-pick ecbuild changes on top). 

First, is that something others are OK with, and do we need to discuss which CICE-Consortium/Icepack tag to update to?

Second, how do we do this better? Manually update more often? (I don't mind doing this myself every once in a while). Forget about this repo, the ecbuild wrappers and somehow build without them? (I don't know how).

@guillaumevernieres, @travissluka, @Dooruk.

**Comment by Dooruk** [comment]
I asked Bin Zhao (GMAO CICE expert) and he said typically there has been two major updates (for both CICE+Icepack) that we have been keeping up with. However, that cadence will drop as there has been some changes/retirement within the CICE community. So we could bump Icepack commit once a year with a commit that works for everyone. Would UFS require more frequent updates than that?

Not sure about the ecbuild suggestion but happy to accommodate if it's easier.

This is the [GEOS-ESM fork](https://github.com/GEOS-ESM/Icepack) we use, looks like we are about 6 months behind.

**Comment by travissluka** [comment]
@shlyaeva @guillaumevernieres I do see that there are some things in our fork that were changed to public (e.g. `init_vertical_profile`). Is it still the case with the newer icepack version that we would need to make those changes? If so, I'm fine with just updating our fork to a newer upstream version, pick whichever one you want.

In the long run, though, I would prefer that we don't maintain our own fork of icepack just for ecbuild purposes. I'm not sure what the best solution would be, but I imagine either 1) putting icepack in spack-stack or 2) having icepack as a submodule of soca and just keeping a cmake build file (similar to what has been done for mom6).

**Comment by shlyaeva** [comment]
> @shlyaeva @guillaumevernieres I do see that there are some things in our fork that were changed to public (e.g. init_vertical_profile). Is it still the case with the newer icepack version that we would need to make those changes? If so, I'm fine with just updating our fork to a newer upstream version, pick whichever one you want.

I'll wait for Guillaume to confirm once he's back, but I'm pretty sure we don't need those to be public (at least not in soca).

**Comment by shlyaeva** [comment]
> In the long run, though, I would prefer that we don't maintain our own fork of icepack just for ecbuild purposes. I'm not sure what the best solution would be, but I imagine either 1) putting icepack in spack-stack or 2) having icepack as a submodule of soca and just keeping a cmake build file (similar to what has been done for mom6).

I agree. How about the second solution since it's already used for mom6? Would you have time to set up the cmake build file + icepack in the soca repo or give me some pointers on how to do this (I'm not very efficient/good at this)? I'd be happy then to also update the code to the newer version (some subroutines used in Soca2Cice changed signature in the last two years).

**Comment by travissluka** [comment]
@shlyaeva yes, I'll start a branch that uses the icepack as a submodule

**Comment by shlyaeva** [comment]
@travissluka is updating soca to build a limited icepack within soca build: https://github.com/JCSDA-internal/soca/pull/1116.
@Dooruk and @travissluka Once that's merged, I'd like to update the version of icepack that's built within soca to https://github.com/CICE-Consortium/Icepack/commits/3792520561cf9419082ef41f9f0dffd03edf2e43/. It's a bit newer than the version GMAO currently uses with GEOS, but it would only be used for Soca2Cice. I've looked at the difference between the commit that GMAO uses and the one we'd like to use. There are differences in cleanup_itd that we use in Soca2Cice, but they are mostly mechanical (removing of integer arguments that can be inferred from the arrays) and touching on the tracers that we don't use. Are you OK with updating to the commit I propose?

**Comment by travissluka** [comment]
:+1:

**Comment by travissluka** [comment]
fyi, i moved this issue to soca... since the icepack repo will be archived soon

**Comment by Dooruk** [comment]
>  Are you OK with updating to the commit I propose?

Sure, the differences sound minuscule. Thanks for checking!
